### PR TITLE
feat: add IFU functionality

### PR DESCRIPTION
**Pull request by civsiv**
This PR adds IndividualFacilityUse functionality to Server.NET.

In order to fully finish off this PR the following needs to happen (apart from the review comments below):
1. an appsettings.ifu.json file needs to be added to this PR (or based on the comment below re default FU vs IFU behaviour, an appsettings.fuNoIfu.json)
2. In Test Suite, an env file needs to be added to run Test Suite in IFU mode
3. This branch needs to be renamed to be `coverage/ifus` so that the Test Suite branch in 2 run tests against this branch
4. Add individual facility uses to the Github Actions test matrix so that every CI run also runs IFU tests

### CODE REVIEW

File: Examples/BookingSystem.AspNetCore/Stores/FacilityStore.cs [hunk 1]
```diff
@@ -304,6 +308,79 @@ protected override async Task GetOrderItems(List<OrderItemContext<FacilityOpport
...
+                        FacilityType = new List<Concept> {
+                                    new Concept
+                                    {
+                                        Id = new Uri("https://openactive.io/facility-types#a1f82b7a-1258-4d9a-8dc5-bfc2ae961651"),
+                                        PrefLabel = "Squash Court",
+                                        InScheme = new Uri("https://openactive.io/facility-types")
+                                    }
+                                }
+                    };
+                }
```
Comment (nickevansuk) [inline Examples/BookingSystem.AspNetCore/Stores/FacilityStore.cs:382]: The `FacilityUse` definition seems to be duplicated here? Should we define it as a var above the `if` and then just include it in either case? More maintainable that way

File: Fakes/OpenActive.FakeDatabase.NET/Models/FacilityUseTable.cs [hunk 2]
```diff
@@ -13,5 +23,19 @@ public class FacilityUseTable : Table
...
+        //{
+        //    get
+        //    {
+        //        return JsonConvert.DeserializeObject<List<IndividualFacilityUse>>(IndividualFacilityUseString);
+        //    }
+        //    set
+        //    {
+        //        IndividualFacilityUseString = JsonConvert.SerializeObject(IndividualFacilityUses);
+        //    }
+        //}
```
Comment (nickevansuk) [inline Fakes/OpenActive.FakeDatabase.NET/Models/FacilityUseTable.cs:38]: Great if this isn't required - in which case shall we remove it?
```suggestion
```

**Review by nickevansuk — CHANGES_REQUESTED** [review]
In addition to suggested changes previously discussed

**Comment by lukehesluke** [comment]
As mentioned here: https://github.com/openactive/OpenActive.Server.NET/pull/201#issuecomment-1638371372, this IFU PR includes the changes from https://github.com/openactive/OpenActive.Server.NET/pull/201. So, if this IFUs PR gets merged first, please could the merger then close https://github.com/openactive/OpenActive.Server.NET/pull/201? Thanks

**Comment by nickevansuk** [comment]
As per this PR's description, a new PR has been created #204 in order to rename the branch to `coverage/ifus`

**Pull request closed without merging**